tests/test_peak_properties.py

```diff
@@ -0,0 +1,98 @@
+import strax
+import numpy as np
+from hypothesis import given, strategies, example, settings
+import tempfile
+
+
+@settings(max_examples=100, deadline=None)
+@given(
+    # number of peaks
+    strategies.integers(min_value=0, max_value=20),
+    # length of the data field in the peaks
+    strategies.integers(min_value=2, max_value=20),
+)
+def test_index_of_fraction(peak_length, data_length):
+    """
+    Test strax.index_of_fraction
+    """
+    # Setup the peaks
+    dtype = [(('Start time since unix epoch [ns]', 'time'), np.int64),
+             (('Exclusive end time since unix epoch [ns]', 'endtime'), np.int64),
+             (('length of p', 'length'), np.int16),
+             (('area of p', 'area'), np.float64),
+             (('data of p', 'data'), (np.float64, data_length))]
+    peaks = np.zeros(peak_length, dtype=dtype)
+    peaks['time'] = np.arange(peak_length)
+    peaks['endtime'] = peaks['time'] + 1
+
+    # Fill the peaks with random length data
+    for p in peaks:
+        length = np.random.randint(0, data_length)
+        p['length'] = length
+        wf = np.random.random(size=data_length)
+        p['data'] = wf
+
+        for i in range(length, len(wf)):
+            # Set all the out of length data to zero
+            wf[i] = 0.
+
+        if len(peaks):
+            # Compute sum area
+            peaks['area'] = np.sum(peaks['data'], axis=1)
+
+    fraction_desired = np.random.random(size=peak_length)
+    res = strax.index_of_fraction(peaks, fraction_desired)

```

Comment:
Why? I must be overlooking something but this can differ on a peak by peak basics:
https://github.com/AxFoundation/strax/blob/2085f81c227e9516eb68794277a3cc4c2b8898bf/strax/processing/peak_properties.py#L24

Reply:
Ah okay nvm, thanks